bittensor/utils/version.py

```diff
@@ -0,0 +1,89 @@
+from typing import Optional
+from pathlib import Path
+import time
+
+import bittensor
+import requests
+
+VERSION_CHECK_THRESHOLD = 86400
+
+
+def _get_version_file_path() -> Path:
+    return Path.home() / ".bittensor" / ".version"
+
+
+def _get_version_from_file(version_file: Path) -> Optional[str]:
+    if not version_file.exists():
+        bittensor.logging.debug("No bitensor version file found")
+        return None
+
+    mtime = version_file.stat().st_mtime
+    bittensor.logging.debug(f"Found version file, last modified: {mtime}")
+    diff = time.time() - mtime
+
+    if diff >= VERSION_CHECK_THRESHOLD:
+        bittensor.logging.debug("Version file expired")
+        return None
+
+    try:
+        return version_file.read_text()
+    except Exception as e:
+        bittensor.logging.error(f"Failed to read version file: {e}")
+        return None
+
+
+def _get_version_from_pypi(timeout: int = 15) -> str:
+    try:
+        bittensor.logging.debug(
+            f"Checking latest Bittensor version at: {bittensor.__pipaddress__}"
+        )
+        response = requests.get(bittensor.__pipaddress__, timeout=timeout)
+        latest_version = response.json()["info"]["version"]
+
+    except requests.exceptions.Timeout:
+        bittensor.logging.error("Version check failed due to timeout")
+    except requests.exceptions.RequestException as e:
+        bittensor.logging.error(f"Version check failed due to request failure: {e}")
+    else:
+        return latest_version
+
+    raise
+
+
+def get_latest_version(timeout: int = 15) -> str:
+    version_file = _get_version_file_path()
+    latest_version = _get_version_from_file(version_file)
+
+    if not latest_version:
+        try:
+            latest_version = _get_version_from_pypi(timeout)
+        except Exception:
```

Comment:
Maybe at least `RequestException`? Bare `Exception`s are discouraged

Reply:
Here, the actual exception is being handled inside `_get_version_from_pypi`. This block is intended to ensure that the application doesn't fail if something goes wrong during the function call. It can be something else, e.g. malformed json response, which will not be caught by `RequestException`.